cert.go

```diff
@@ -471,10 +493,12 @@ func ValidityBounds(wd depot.Depot, name string) (time.Time, time.Time, error) {
 // DeleteOnExpiration deletes the given certificate from the depot if it has an
 // expiration date within the duration `after`. True is returned if the
 // certificate is deleted, false otherwise.
-func DeleteOnExpiration(wd depot.Depot, name string, after time.Duration) (bool, error) {
+func DeleteOnExpiration(wd Depot, name string, after time.Duration) (bool, error) {
 	var deleted bool
 
-	if !depot.CheckCertificate(wd, name) {
+	if exists, err := CheckCertificateWithError(wd, name); err != nil {
+		return deleted, err
```

Comment:
Wrap error.

Same question as above - should this just return false since deleted hasn't even been set yet?